mesonbuild/modules/gnome.py

```diff
@@ -406,12 +406,22 @@ def _get_gir_dep(self, state):
             kwargs = {'native': True, 'required': True}
             holder = self.interpreter.func_dependency(state.current_node, ['gobject-introspection-1.0'], kwargs)
             self.gir_dep = holder.held_object
-            if self.gir_dep.type_name == 'pkgconfig':
-                self.giscanner = ExternalProgram('g_ir_scanner', self.gir_dep.get_pkgconfig_variable('g_ir_scanner', {}))
-                self.gicompiler = ExternalProgram('g_ir_compiler', self.gir_dep.get_pkgconfig_variable('g_ir_compiler', {}))
+            giscanner = state.environment.lookup_binary_entry(MachineChoice.HOST, 'g-ir-scanner')
+            if giscanner is not None:
+                self.giscanner = ExternalProgram.from_bin_list(state.environment, MachineChoice.HOST, 'g-ir-scanner')
```

Comment:
This does the lookup twice, better do `ExternalProgram.from_entry('g-ir-scanner', giscanner)` here, right?

Reply:
Hmm, not sure but I used [this](https://github.com/mesonbuild/meson/blob/c53b6379597be5961b4e69e7f187608452874e4c/mesonbuild/environment.py#L574-L577) as an example.

Reply:
Updated with your suggestion.